dashboard/modules/runtime_env/runtime_env_agent.py

```diff
@@ -200,64 +184,74 @@ def __init__(self, dashboard_agent):
         _initialize_internal_kv(self._dashboard_agent.gcs_client)
         assert _internal_kv_initialized()
 
-        self._pip_manager = PipManager(self._runtime_env_dir)
-        self._conda_manager = CondaManager(self._runtime_env_dir)
-        self._py_modules_manager = PyModulesManager(self._runtime_env_dir)
-        self._java_jars_manager = JavaJarsManager(self._runtime_env_dir)
-        self._working_dir_manager = WorkingDirManager(self._runtime_env_dir)
+        self._pip_plugin = PipPlugin(self._runtime_env_dir)
+        self._conda_plugin = CondaPlugin(self._runtime_env_dir)
+        self._py_modules_plugin = PyModulesPlugin(self._runtime_env_dir)
+        self._java_jars_plugin = JavaJarsPlugin(self._runtime_env_dir)
+        self._working_dir_plugin = WorkingDirPlugin(self._runtime_env_dir)
         self._container_manager = ContainerManager(dashboard_agent.temp_dir)
 
+        self._base_plugins = [
+            self._working_dir_plugin,
+            self._pip_plugin,
+            self._conda_plugin,
+            self._py_modules_plugin,
+            self._java_jars_plugin,
+        ]
+        self._uri_caches = {}
+        self._base_plugin_cache_managers = {}
+        for plugin in self._base_plugins:
+            # Set the max size for the cache.  Defaults to 10 GB.
+            cache_size_env_var = f"RAY_RUNTIME_ENV_{plugin.name}_CACHE_SIZE_GB".upper()
+            cache_size_bytes = int(
+                (1024 ** 3) * float(os.environ.get(cache_size_env_var, 10))
+            )
+            self._uri_caches[plugin.name] = URICache(
+                plugin.delete_uri, cache_size_bytes
+            )
+            self._base_plugin_cache_managers[plugin.name] = PluginCacheManager(
```

Comment:
Hmm, I think we don't want `Plugin` class authors to have implement the caching logic themselves, since it's the same (or similar) for each plugin.  That's why instead we have the PluginCacheManager class that wraps the user's plugin class and handles the caching.

Unless you're just talking about the `uri_caches` being exposed to the file?  I had thought about that one for a bit.  Right now the ReferenceTable class needs direct access to all the URI caches of all the plugins.  The alternative is to make the `uri_cache` a public attribute of the plugin cache manager class.  So in the ReferenceTable, instead of `self._uri_caches[plugin.name]` we can access it via `self._base_plugin_cache_managers[plugin.name].uri_cache`.  Would this be preferable?